File: Go/0014-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}

	result := ""

	for j := 0; j < len(strs[0]); j++ {
		c := strs[0][j]
		for i := 1; i < len(strs); i++ {
			if j >= len(strs[i]) || strs[i][j] != c {
				return result
			}
		}
		result += string(c)
	}

	return result
}

File: Rust/0014-longest-common-prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        if strs.is_empty() {
            return "".to_string();
        }

        let mut result = String::new();
        let first = strs[0].as_bytes();

        for j in 0..first.len() {
            let c = first[j];

            for i in 1..strs.len() {
                let current = strs[i].as_bytes();
                if j >= current.len() || current[j] != c {
                    return result;
                }
            }

            result.push(c as char);
        }

        result
    }
}
